exchanges/binanceus/binanceus_websocket.go

```diff
@@ -0,0 +1,1093 @@
+					err := json.Unmarshal(rawData, &bo)
+					if err != nil {
+						return fmt.Errorf("%v - Could not convert to bookOrder structure %s ", err, bi.Name)
+					}
+					pair, err := currency.NewPairFromFormattedPairs(bo.S, pairs, format)
+					if err != nil {
+						return err
+					}
+					bo.Symbol = pair
+					bi.Websocket.DataHandler <- &bo
+					return nil
+				case "aggTrade":
+					var agg WebsocketAggregateTradeStream
+					err := json.Unmarshal(rawData, &agg)
+					if err != nil {
+						return fmt.Errorf("%v - Could not convert to aggTrade structure %s ", err, bi.Name)
+					}
+					bi.Websocket.DataHandler <- agg
+					return nil
+				default:
+					bi.Websocket.DataHandler <- stream.UnhandledMessageWarning{
+						Message: bi.Name + stream.UnhandledMessage + string(respRaw),
+					}
+				}
+			}
+		} else if wsStream == "!bookTicker" {
+			var bt OrderBookTickerStream
+			if data, ok := multiStreamData["data"]; ok {
+				rawData, err := json.Marshal(data)
+				if err != nil {
+					return err
+				}
+				pairs, err := bi.GetEnabledPairs(asset.Spot)
+				if err != nil {
+					return err
+				}
+
+				format, err := bi.GetPairFormat(asset.Spot, true)
+				if err != nil {
+					return err
+				}
+				err = json.Unmarshal(rawData, &bt)
+				if err != nil {
+					return fmt.Errorf("%v - Could not convert to bookOrder structure %s ", err, bi.Name)
+				}
+				pair, err := currency.NewPairFromFormattedPairs(bt.S, pairs, format)
+				if err != nil {
+					return err
+				}
+				bt.Symbol = pair
+				bi.Websocket.DataHandler <- &bt
+				return nil
+			}
+		}
+	}
+	return fmt.Errorf("unhandled stream data %s", string(respRaw))
+}
+
+// UpdateLocalBuffer updates and returns the most recent iteration of the orderbook
+func (bi *Binanceus) UpdateLocalBuffer(wsdp *WebsocketDepthStream) (bool, error) {
+	enabledPairs, err := bi.GetEnabledPairs(asset.Spot)
+	if err != nil {
+		return false, err
+	}
+
+	format, err := bi.GetPairFormat(asset.Spot, true)
+	if err != nil {
+		return false, err
+	}
+
+	currencyPair, err := currency.NewPairFromFormattedPairs(wsdp.Pair,
+		enabledPairs,
+		format)
+	if err != nil {
+		return false, err
+	}
+
+	err = bi.obm.stageWsUpdate(wsdp, currencyPair, asset.Spot)
+	if err != nil {
+		init, err2 := bi.obm.checkIsInitialSync(currencyPair)
+		if err2 != nil {
+			return false, err2
+		}
+		return init, err
+	}
+
+	err = bi.applyBufferUpdate(currencyPair)
+	if err != nil {
+		bi.flushAndCleanup(currencyPair)
+	}
+
+	return false, err
+}
+
+// GenerateSubscriptions generates the default subscription set
+func (bi *Binanceus) GenerateSubscriptions() ([]stream.ChannelSubscription, error) {
+	var channels = []string{"@ticker", "@trade", "@kline_1m", "@depth@100ms"}
+	var subscriptions []stream.ChannelSubscription
+
+	pairs, err := bi.GetEnabledPairs(asset.Spot)
+	if err != nil {
+		return nil, err
+	}
+
+	for y := range pairs {
+		for z := range channels {
+			lp := pairs[y].Lower()
+			lp.Delimiter = ""
```

Comment:
BinanceUS has a subscription limit of 1024 streams per connection as per https://docs.binance.us/#websocket-rate-limits

I found when running BinanceUS with all pairs enabled, this would cause a "too many subscriptions" error loop for the websocket.
So I suggest the following:

```go

subs:
	for y := range pairs {
		for z := range channels {
			lp := pairs[y].Lower()
			lp.Delimiter = ""
			if len(subscriptions) >= 1023 {
				log.Warnf(log.WebsocketMgr, "BinanceUS has 1024 subscription limit. Requested %v, using the first 1024", len(pairs)*len(channels))
				break subs
			}
			subscriptions = append(subscriptions, stream.ChannelSubscription{
				Channel:  lp.String() + channels[z],
				Currency: pairs[y],
				Asset:    asset.Spot,
			})
		}
	}
```

To test this yourself, you can add all pairs to your enabled pairs in config, or to temporarily test, just use `go run . --enableallpairs`

After this is fixed, I think everything is good for me